tensorzero-core/src/tool.rs

```diff
@@ -1142,29 +1710,67 @@ mod tests {
             ToolChoice::Specific("get_temperature".to_string());
     }
 
+    // Helper to create ClientSideFunctionTool quickly
+    fn make_client_side_tool(
+        name: &str,
+        description: &str,
+        parameters: Value,
+        strict: bool,
+    ) -> ClientSideFunctionTool {
+        ClientSideFunctionTool {
+            name: name.to_string(),
+            description: description.to_string(),
+            parameters,
+            strict,
+        }
+    }
+
+    // Helper to construct ToolCallConfigConstructorArgs with defaults
+    fn make_tool_call_config_args<'a>(
```

Comment:
How about
```rust
impl ToolCallConfigConstructorArgs {
  pub fn with_dynamic_tool_params() {
    ConstructorArgs {
      // fill in the tool param fields,
      ..default()
    }
  }
}
```

Then you write this at callsite:
```rust
ConstructorArgs {
  function_tools,
  ...
  ..ConstructorArgs::with_dynamic_tool_params()
}
```

which I think reads better than a function with 5 unnamed arguments.